pandas/io/parsers/base_parser.py

```diff
@@ -930,7 +939,29 @@ def _check_data_length(
                 stacklevel=find_stack_level(),
             )
 
-    def _evaluate_usecols(self, usecols, names):
+    @overload
+    def _evaluate_usecols(
+        self, usecols: Callable, names: Sequence[Hashable]
+    ) -> set[int]:
+        ...
+
+    @overload
+    def _evaluate_usecols(
+        self, usecols: set[int], names: Sequence[Hashable]
```

Comment:
You can probably combine the first two overloads by having `usecols: Callable | set[int]`.

Probably can also replace `Callable` with `Callable[[Hashable] int]`.

Reply:
Not that familiar with Callable annotations. Would this cover something like ``lambda x: x % 4``?

Combined the overloads

Reply:
Yes, `Callable[[Hashable], int]` covers `lambda x: x % 4` :)

Reply:
Thx, added